nesta/core/routines/datasets/nih/nih_dedupe_task.py

```
@@ -1,131 +1,75 @@
-        for count, batch in enumerate(batches, 1):
-            # Magical '0.3' is the lower end of the deduplication
-            # fraction found by inspection
-            done = sum(_id in done_ids
-                       for _id in batch) / len(batch) > 0.3
-            # write batch of ids to s3
-            batch_file = ''
-            if not done:
-                batch_file = put_s3_batch(batch,
-                                          self.intermediate_bucket,
-                                          self.routine_id)
-            params = {
-                "batch_file": batch_file,
-                "config": 'mysqldb.config',
-                "bucket": self.intermediate_bucket,
-                "done": done,
-                'outinfo': es_config['host'],
-                'out_port': es_config['port'],
-                'out_index': es_config['index'],
-                'in_index': _old_config['index'],
-                'out_type': es_config['type'],
-                'aws_auth_region': es_config['region'],
-                'entity_type': 'paper',
-                'test': self.test,
-                'routine_id': self.routine_id
-            }
 
-            job_params.append(params)
-            if self.test and count > 1:
-                logging.warning("Breaking after 2 batches "
-                                "while in test mode.")
-                logging.warning(job_params)
+class RootTask(luigi.WrapperTask):    
+    date = luigi.DateParameter(default=dt.now())
+    production = luigi.BoolParameter(default=False)
+    
+    def requires(self):
+        for source, orm in (("phr", PhrVector),
+                            ("abstract", AbstractVector)):
+            yield VectorDedupeTask(vector_orm=orm,
+                                   source=source,
+                                   date=self.date,
+                                   test=not self.production)
+            if not self.production:
                 break
```

Could this mask errors in deduping abstracts when testing?

---

the main reason for this is that it's quite a slow turnaround time: but you would only expect errors to crop up if there are e.g. weird formattings etc in abstracts compared with PHR, but having already processed *everything* in `production` mode in anticipation of this PR I can already say that there really aren't any edge cases of that type. For me, the purpose of the `test/production` modes is something between unit and end-to-end testing, to at least check that the basic principle of the pipeline is working - which I think is covered in this case as `PhrVector` and `AbstractVector` are so similar.